### DISCUSSION TITLE
How the outfit scanning changes affect missions

### DISCUSSION THREAD

**Discussion by junetried** [endless-sky/endless-sky#8493] [Balance, Mechanics, and Interactions]
Outfit scanning was recently overhauled apparently, and with it scanning is overall much slower. There are a couple missions that make you scan hostile ships, and I'm concerned that they might need to be rebalanced.

There's a Remnant mission to scan a Tubfalet in Kor Zena'i. Since the outfit scan change, this mission seems a lot harder to actually complete. Scanning it takes far too long considering both its firepower and the fact that it is equipped with two banisher turrets. Disabling it is also very difficult, since it's in hostile Korath space and you're likely to be swarmed if you stay long enough to disable and then scan a large warship. I eventually scanned it by taking a single Dreadnought equipped with twenty-three outfit scanners.

There's also (if I remember correctly) a Free Worlds mission to scan a Republic gunboat equipped with electron beams, and that one is probably also problematic for similar reasons.

**Comment by Zitchas** [comment]
Thank you for the feedback. It's very useful for us to get the lived experiences of players actually playing the game.

While part of the purpose of the scanner overhaul was to make it more of a mini-game and less of a no-effort exercse, and thus some optimization for scanning is expected to be necessary, needing 23 outfit scanners is excessive.

I have a PR open ( https://github.com/endless-sky/endless-sky/pull/8395 ) to give the Remnant more reasonable levels for their environment and the challenges (such as this listed mission) that they are expected to be doing, however that PR has stalled for various reasons. As such, it's unknown when the fix will actually be applied.

**Comment by ziproot** [comment]
There is also the one with scanning the Republic ships. It was not my 23 outfit scanners that did the trick (the scan time barely went down from 1 to 23 so I don't even think all 23 were necessary; I was just using as much as I could), but the atomic engines that let me keep up with the ships so I could keep scanning them. Last time I checked the PR buffing scanning again the problem disappeared. 
EDIT: Scan time went down, not up
EDIT 2: The PR mentioned is #8288. If you have ESLauncher2 you can try PRs. I would recommend testing that to see if there's improvements if you have ESLauncher2.
EDIT 3: If you do not have ESLauncher2 you can still try the PR by going to [checks](https://github.com/endless-sky/endless-sky/pull/8288/checks), clicking the "artifacts" drop down menu in the top right, then choosing the file based on your operating system.

**Comment by IsocratesGrey** [comment]
Part of this seems to be with the way that scanners stack. As in the changes were made to make scanning a bit more difficult, but for those with only a single scanner (which should be the normal method) it's now virtually impossible. Perhaps some way of having diminishing returns from additional scanners? That way it makes multiple scanners instead of a single scanner of the best type you can get a waste of money.

**Reply by ziproot** [reply]
I thought scanners already provide diminishing returns.

**Reply by junetried** [reply]
> I thought scanners already provide diminishing returns.

I know that's how ramscoop is calculated. I was under that impression too.

**Reply by Zitchas** [reply]
They did, and do, operate the same way as Ramscoops, where one needs 4x the ramscoop value to get 2x the performance.

With the new system, however, there's also a 50% dropoff at max range. So you only get "Full speed", whatever that happens to be for your configuration and target, when you are sitting on top of them. At max range your scan will only progress at 50% of that speed. On top of that, the more mass your target has, the slower it will go as well. So if you're trying to scan something that is both very big and very agile, it'll both take longer and make you work a lot harder to get it.

Of course, making it actually take effort instead of just "be within range and hit "scan" for a few seconds" was exactly the point of the changes, so that was a success. And overall I think it was the right direction. It just needs some tweaking to make it a good minigame.

**Reply by junetried** [reply]
I don't necessarily think the scanning needs changed much, overall I think I like the change. I think the mission I mentioned should probably place the ship in the Ember Wastes or an otherwise not so hostile system. That would make it much more feasible to disable the ship before scanning it, which sounds like the more reasonable route anyway.

**Reply by Zitchas** [reply]
Interestingly I just did that mission, and that is what I did. I couldn't get a scan, or rather, I was going to die before it finished, so I lured it out into the outskirts, then disabled it, scanned it, looted it, and left.

**Reply by IsocratesGrey** [reply]
The missions I'm finding became much less fun was "FW Recon 1" and "FW Recon 2". They are supposed to be completable with an early game ship like a Clipper, Scout, Bounder, or Freighter and a single outfit scanner. Because you need to basically be able to match speed with the cruisers this can be very difficult. For those with mid game ships or alien ships I'm sure those missions remain easy.

I don't want it reverted, I like the idea behind the mini game. What I don't like is that it breaks the immersion of the FW Prologue because a single scanner vs a cruiser means that you need to be right on top of a ship that can move faster than you for 5-10 seconds. If it's supposed to be an early game mission for a pilot in their second or third ship then scanners need to be adjusted for that.

On the flip side, it also seems to have made it much easier for me to smuggle goods, which probably is good for pirate players, but again breaks immersion because smuggling is no longer a high risk-high reward activity.

Suggestions: slight improvement in the drop off at max range combined with a faster diminishing return, so at max range it drops off 25% but it takes 6x ramscoop value for 2x performance. Alternatively instead of working the diminishing returns angle, simply have it only use a single scanner (whichever has the highest value).

**Reply by junetried** [reply]
Maybe at minimum range (being on top of the ship), the scanner should have a speed more like 125% or 150%?

**Reply by Hecter94** [reply]
> They are supposed to be completable with an early game ship like a Clipper, Scout, Bounder, or Freighter and a single outfit scanner. 

What makes you think you're supposed to use only a single outfit scanner for those missions? Is that mentioned in dialogue or something? 

The intent is that you install scanners based on what you need. If you're in a small, agile ship scanning another small ship, one might be enough. 
Cruisers aren't a small ship, so one probably isn't enough.

If there's dialogue that implies you should only install one, that should probably be updated to reflect the new scanning formula.

**Reply by Zitchas** [reply]
Worth noting that the problem might not actually be with the scanner at all; but rather the fact that some ships (like the cruiser) have only a tiny fraction of the mass that they should actually have. With a proper amount of mass, they wouldn't be changing directions like a fighter, and thus they'd a lot easier to stay close to.

Regarding the stacking thing, it's numerical change, not actually about the literal stacking of outfits. If you have scan power 10, then to get double the range you need to have scan power 40. It doesn't matter if you got the original 10 in one outfit or five; nor does it matter if you get the 40 using one outfit or twenty. It's just that to get double the range, you need quadruple the amount of scan power.

**Reply by junetried** [reply]
> What makes you think you're supposed to use only a single outfit scanner for those missions? Is that mentioned in dialogue or something?

This mission is meant to be doable using a very early ship like one of those, which won't have the space for many scanners without compromising potentially many normal outfits. For the slower ships, like the freighter-class ships, you also have the problem of needing to make space for more powerful engines just to keep up with the Navy Cruiser.

**Reply by IsocratesGrey** [reply]
Regarding the stacking, what I meant originally by wanting diminishing returns isn't pure scan power, but how the game adds multiple scanners together. So for example, 1 scanner of power 10 should be substantially better than 10 scanners of power 1. This balances the scanner mini game better in alignment with tech access.

**Reply by IsocratesGrey** [reply]
> If there's dialogue that implies you should only install one, that should probably be updated to reflect the new scanning formula.

The dialogue says "The oufitter [SIC] stocks scanners if you don't have one, though you might want to install a few of them to improve performance". This is new dialogue added in 0.10.0. In the current live version the recommendation to add more than a single scanner wasn't there.

Each is mass 2, power 25, efficiency 15, cost 24000.

Because it's in the prologue the expectation shouldn't be that the player can load more than two or three at the most on their ship at the time they get this mission, as in early game many new players may not have begun min-maxing their ships yet, and shouldn't be expected to have alien outfits yet.

Edit: The mission pays 10,000 for the first mission and 50,000 for the second mission. Which means if the mission requires more than two scanners to complete then it's a net loss for the player for the purpose of progression in the prologue.

Again, I like the change to a mini-game, but it needs some tweaking to not pull the player out of the immersion from frustration.

**Reply by IsocratesGrey** [reply]
Because I realize I've taken this thread off topic a little, I created a thread #8520 that is more broadly around scanner changes impacting gameplay instead of this specific Remnant mission.

**Comment by EjoThims** [comment]
These are specifically non-combat missions.

If you need to drop some shields or weapons or even buy an outfit expansion to stack enough scanners to make it comfortable alongside the rest of your loadout, that teaches new players how to begin playing that min/max game.

It's also quite possibly the first time the player may be confronted with the mission push to upgrade their ship from very starting vessels (especially since unless they've been upgrading in the Deep themselves or way laying pirates, there's no way to their engines are as good), so it definitely shouldn't be going back to anywhere near as easy to do with a single scanner from a shuttle or sparrow as it used to be.

**Reply by IsocratesGrey** [reply]
No, I like the shift to making scanning more difficult, but I also don't like the idea that it's a gameplay mechanic the player is encouraged to brute force by stacking as many scanners as they can. Instead it should be about being able to stay in proximity to what they are scanning for a reasonable amount of time. But right now the proximity/time/power balance is wrong. So I don't pull this discussion further off topic I'll move over to #8520 .

**Reply by junetried** [reply]
The Tubfalet is a Korath Exile ship, and it is hostile. Completing it without combat requires some absurd setup, like a Dreadnought with 23 outfit scanners.
